### ISSUE TITLE
Facets contain duplicates and incorrect items

### ISSUE THREAD

**Issue by Castronova** [hydroshare/hydroshare#4035]
The facets on the discover page are not curated. This is problematic because it results in both duplicate and incorrect values in the facets. For example, some names appear multiple times in the "Author" filter because a name may be associated with more than one account. Similarly, some names in the "Author" filter are not human names. This is because users are allowed to enter any value in the author field. Examples are shown below.

![image](https://user-images.githubusercontent.com/4822372/98886558-55cfaa80-2462-11eb-9211-26095c11ee7c.png)

![image](https://user-images.githubusercontent.com/4822372/98886605-65e78a00-2462-11eb-9b4d-48c063b6efed.png)

**Comment by cuahsimarko** [comment]
This was true in the old Discover and less obvious, because it was not sorting by name. Design is needed and perhaps rethinking person objects, as there is no way to know currently if it's simply the same name, "John Smith" or indeed an identical person. If different accounts are used, is there a benefit to having the separation of resources, since they could be for different organizations or other distinctions.

**Comment by alvacouch** [comment]
@Castronova  It is possible to deal with duplicate names via the way the records are faceted. What I presume is that the "duplicates" are really slightly different depictions of the same name, which could be repaired by normalization to last name, first initial. 

I have no solution for misspellings! 

The other issue of garbage in the metadata is deeper and may lead us to not exposing facets based upon uncontrolled vocabularies at all.

**Comment by alvacouch** [comment]
@dtarb @castronova @cuahsimarko Should we try last name/first initial to resolve this, or is that worth doing?

**Comment by cuahsimarko** [comment]
There's no way of definitively distinguishing between identical names. I think this should be closed with no action, but I can make a note of alerting at profile name entry/edit that it matches a duplicate name/account. @Castronova as the ticket creator can you please comment/close?

**Comment by dtarb** [comment]
@alvacouch I do not think it is worth doing lastname/first initial to attempt to resolve this.  I would vote for closing this now, but putting as a discussion item more thoroughly addressing the representations for people and organizatoin names, aka party's or the party model.

Here are some ideas.  
1. It is the responsibility of resource creators and editors to make sure that information for the authors and contributors is correct and complete.  In resource pages we should just display information as it is in the resource records.
2. Parties should be disambiguated using unique identifiers.  In HydroShare email addresses, as well as identifiers such as ORCID can be part of a user profile and can be added to a creator/contributor record.  Selecting a HydroShare user to add to a resource creator/contributor record copies this information.  
3. The discovery database and facets should group together authors that are known to be the same party by virtue of having the same unique identifier.  This would resolve the Castronova repeat illustrated above.  Different firstname and lastname variants could be regarded as aliases and discovery of any of them discovers all.  The data representation to achieve this may not be trivial, hence my suggestion we not do this now.

**Comment by cuahsimarko** [comment]
@dtarb @alvacouch I support the idea of a Unique ID, perhaps not something entered by the user, such as an underlying uuid4 (random identifier) much like the database best practice with an auto incrementing/generated index. Respective systems could then use the UID for retrieval. The final step would be providing a method for resource creators and editors to consolidate their accounts under one UID and a campaign to call users to action.

**Comment by horsburgh** [comment]
I vote against trying to do additional disambiguation of things like Authors based on some sort of additional identifier. Authors that are HydroShare users already have a unique identifier in the system, but not all authors are users. The real problems are with authors that are not HydroShare users because their information does not get copied from their profile (or their information gets copied and then they change their profile). For instance, how would you link up the same author name entered by two different users (e.g., two data managers creating HydroShare resources for other people) at two different times? You wouldn't unless the exact same information - name, email address, etc. - is entered at both times, which obviously isn't happening now or we wouldn't have this problem.

The discovery facets act as filters - their purpose is to help subset search results. If two authors have the exact same name, you get both of their results when you click on that name. This isn't a problem - it just forces the user to sift through a longer list of results. A random unique identifier for each author having the same name won't help a user tell the difference between the two. User's won't memorize unique identifiers.

Even if we try to consolidate names through grouping, there are instances where grouping should not happen - e.g., a person may get married and change their name - it's the same person, but they published under two different names. Also - If you group two names together that are different (the Castronova example above), which one gets displayed?

There are limitations to having a field like Author as a facet. Given the large number of authors and the possibility of duplicates and alternative spellings, I would ask whether Author should even be a facet?

The filters implemented by the DataONE project on their discovery page seem much more flexible and much less problematic: https://search.dataone.org/data

**Comment by cuahsimarko** [comment]
@dtarb @martinseul @horsburgh @alvacouch I would like to close this, per Dave's vote, if others agree. Where can we capture any necessary information that needs to be carried forward? Can a stakeholder please action this?

**Comment by cuahsimarko** [comment]
Closing per thread and can be reopened if deemed necessary.

**Comment by cuahsimarko** [comment]
I will link this to the other duplicate name, facets, first/last/organization issues so stakeholders can consolidate, perhaps.

**Comment by cuahsimarko** [comment]
Possibly related https://github.com/hydroshare/hydroshare/issues/1774 "insufficient metadata to sort creators by last name"
